File: C_Assignment/Assignment_44/programAQ3.c
///////////////////////////////////////////////////////////
//
//  File name : Assignement44Question3.c
//  Description : Write a program which accept matrix and return largest number from both the
//                diagonals
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*

Input :
3 2 5 9
4 3 2 2
8 4 1 5
3 9 7 5

Output : 9
*/

#include<stdio.h>

int MaxDiagonal(int Arr[][4], int iRow, int iCol)
{
    int i = 0;
    int iMax = Arr[0][0];

    for (i = 0; i < iRow; i++)
    {
        if(Arr[i][i] > iMax)
        {
            iMax = Arr[i][i];
        }

        if(Arr[i][iCol - i - 1] > iMax)
        {
            iMax = Arr[i][iCol - i - 1];
        }
    }
    
    return iMax;
}

int main()
{
   int iRow = 0,iCol = 0,i = 0,j = 0;
    int Arr[4][4];
    int iRet = 0;
    
    printf("Enter number of rows :\n");
    scanf("%d",&iRow);

    printf("Enter number of columns :\n");
    scanf("%d",&iCol);

    printf("Enter matrix elements :\n");
    for(i = 0;i < iRow;i++)
    {
        for(j = 0;j < iCol;j++)
        {
            scanf("%d",&Arr[i][j]);
        }
    }
    
    iRet = MaxDiagonal(Arr,iRow,iCol);

    printf("Largest number is : %d\n",iRet);

    return 0;
}

File: Java_Assignment/Assignment_51/programAQ3.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement51Question3.java
//  Description : Write a java program which accept matrix from user and return its
//                transpose matrix.The transpose of a given matrix is formed by
//                interchanging the rows and columns of a matrix.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

import java.util.Scanner;

class Matrix
{
    private int Arr[][];

    public Matrix(int A,int B)
    {
        Arr = new int [A][B];
    }

    public void Accept()
    {
        System.out.println("Please enter the elements of matrix");

        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;

        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0; j< Arr[i].length;j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public void Display()
    {
        System.out.println("Elements of the matrix are : ");

        int i = 0, j = 0;

        for(i = 0;i < Arr.length;i++)
        {
            for(j = 0; j< Arr[i].length;j++)
            {
                System.out.print(Arr[i][j]+"\t");
            }
            System.out.println();
        }
    }

    public void SwapRows()
    {
        int j = 0 ,i = 0, temp = 0;

        for(i = 0;i < Arr.length - 1;i+=2)
        {   
            for(j = 0;j < Arr[i].length;j++)
            {
                temp = Arr[i][j];
                Arr[i][j] = Arr[i + 1][j];
                Arr[i + 1][j] = temp;
            }
        }
    } 

    public void SwapColumns()
    {
        int j = 0 ,i = 0, temp = 0;

        for(j = 0;j < Arr[0].length - 1;j+=2)
        {   
            for(i = 0;i < Arr.length;i++)
            {
                temp = Arr[i][j];
                Arr[i][j] = Arr[i][j + 1];
                Arr[i][j + 1] = temp;
            }
        }
    } 

    public int[][] Transpose()
    {
        int i = 0, j = 0;

        int Trans[][] = new int[Arr[0].length][Arr.length];

        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                Trans[j][i] = Arr[i][j];
            }
        }

        return Trans;
    }

    public void DisplayTranspose(int Brr[][])
    {
        System.out.println("Transpose Matrix : ");

        int i = 0, j = 0;

        for(i = 0; i < Brr.length; i++)
        {
            for(j = 0; j < Brr[i].length; j++)
            {
                System.out.print(Brr[i][j]+"\t");
            }
            System.out.println();
        }
    }

}

public class programAQ3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows : ");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of colums : ");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);

        mobj.Accept();

        System.out.println("\nOriginal Matrix:");
        mobj.Display();

        // mobj.SwapRows();

        // System.out.println("\nMatrix after swapping rows:");
        // mobj.Display();

        // mobj.SwapColumns();

        // System.out.println("\nMatrix after swapping Columns:");
        // mobj.Display();

        int Ret[][] = mobj.Transpose();

        mobj.DisplayTranspose(Ret);

        sobj.close();
    }   
}